src/chain.h

```diff
@@ -414,16 +414,9 @@ class CDiskBlockIndex : public CBlockIndex
         return block.GetHash();
     }
 
-
-    std::string ToString() const
-    {
-        std::string str = "CDiskBlockIndex(";
-        str += CBlockIndex::ToString();
-        str += strprintf("\n                hashBlock=%s, hashPrev=%s)",
-            GetBlockHash().ToString(),
-            hashPrev.ToString());
-        return str;
-    }
+    // To be called only by the base class.
+    uint256 GetBlockHash() = delete;
+    uint256 ToString() = delete;
```

Comment:
Deleted functions may also return `void`

Reply:
It may return anything because it is irrelevant, but if it is going to be different than the return type of the base method, then I would prefer it to return [`std::vector<double>`](https://purple.bikeshed.com/).

Reply:
> Why return `uint256` from `ToString()`? It is irrelevant in this case, but better have the same return type as the parent `CBlockIndex::ToString()` in order to avoid people asking this question :)

Agree, fixed (thanks!)